### PR TITLE
BREAKING CHANGE: Support queue and blob trigger + package refactor [SFEQS-910]

### PR DESCRIPTION
**Pull request by lucacavallaro**
- Support for new Azure Function triggers: Blob Storage (`fromBlobStorage`) and Storage Queue (`fromQueueMessage`)
- Following [RFC 7807](https://datatracker.ietf.org/doc/html/rfc7807), error handling has been completely redesigned and enhanced 
- Now (almost) all decoder perform a runtime validation of the input (context) parameters.

⚠️ **This package is not stable yet, so any PR (like this one) could change the shape of the API.**

### CODE REVIEW

File: src/azure/index.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { HttpResponse } from "../http";
+
+export * from "./http";
+export * from "./queue";
+export * from "./blob";
+
+export const unsafeRun =
+  (handler: Handler<Context>): AzureFunction =>
+  async (ctx) => {
+    const output = await handler(ctx)();
```
Comment (gunzip) [inline src/azure/index.ts:13]: what if handler throws here?

File: src/azure/queue.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import { parse } from "fp-ts/Json";
+
+import { validate } from "../validation";
+import { trigger, InvalidTriggerError } from "./trigger";
+
+const isQueueTriggeredFunctionContext = (
+  ctx: azure.Context
+): ctx is azure.Context & {
+  bindingData: {
+    QueueTrigger: string;
```
Comment (gunzip) [inline src/azure/queue.ts:18]: why not

```suggestion
    queueTrigger: string;
```

File: src/azure/queue.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+export const fromQueueMessage =
+  <T>(schema: t.Decoder<unknown, T>) =>
+  (ctx: azure.Context) =>
+    pipe(
+      ctx,
+      E.fromPredicate(
+        isQueueTriggeredFunctionContext,
+        () =>
+          new InvalidTriggerError(
+            "This function can be triggerd only by a Queue Message"
```
Comment (gunzip) [inline src/azure/queue.ts:36]:
```suggestion
            "This function can be triggered only by a Queue Message"
```

File: src/azure/queue.ts (hunk 2 above)
Comment (lucacavallaro) [reply src/azure/queue.ts:18]: I didn't choose the key, it's microsoft's fault 😢

https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-queue-trigger?tabs=in-process%2Cextensionv5&pivots=programming-language-javascript#message-metadata

inconsistent compared to the other keys...

File: src/azure/index.ts (hunk 1 above)
Comment (lucacavallaro) [reply src/azure/index.ts:13]: The Azure Function crashes with that error, such as when "Error" is returned as output

**Review by grausof — APPROVED** [review]

**Comment by grausof** [comment]
I would add some examples (or tests) of the trigger from **Blob Storage** and **Storage Queue**

**Comment by grausof** [comment]
⚠ The storage blob trigger does not seem to work correctly:

```
[2022-08-01T12:19:23.401Z] Executed 'Functions.ValidateDocumentUpload' (Failed, Id=3a1a388a-dccd-4a5a-b86b-9b5938eed63a, Duration=515ms)
[2022-08-01T12:19:23.402Z] System.Private.CoreLib: Exception while executing function: Functions.ValidateDocumentUpload. System.Private.CoreLib: Result: Failure
[2022-08-01T12:19:23.402Z] Exception: Unable to validate the metadata schema
[2022-08-01T12:19:23.402Z] Stack: Error: Unable to validate the metadata schema
[2022-08-01T12:19:23.402Z]     at /Users/francesco.grauso/Desktop/Firma con IO/io-functions-sign/node_modules/@pagopa/handler-kit/lib/validation.js:41:91
[2022-08-01T12:19:23.402Z]     at /Users/francesco.grauso/Desktop/Firma con IO/io-functions-sign/node_modules/fp-ts/lib/Either.js:532:46
[2022-08-01T12:19:23.402Z]     at /Users/francesco.grauso/Desktop/Firma con IO/io-functions-sign/node_modules/fp-ts/lib/function.js:184:24
[2022-08-01T12:19:23.402Z]     at /Users/francesco.grauso/Desktop/Firma con IO/io-functions-sign/node_modules/@pagopa/handler-kit/lib/azure/blob.js:35:99
[2022-08-01T12:19:23.402Z]     at /Users/francesco.grauso/Desktop/Firma con IO/io-functions-sign/node_modules/fp-ts/lib/Either.js:333:38
[2022-08-01T12:19:23.402Z]     at pipe (/Users/francesco.grauso/Desktop/Firma con IO/io-functions-sign/node_modules/fp-ts/lib/function.js:288:20)
[2022-08-01T12:19:23.402Z]     at /Users/francesco.grauso/Desktop/Firma con IO/io-functions-sign/node_modules/@pagopa/handler-kit/lib/azure/blob.js:34:74
[2022-08-01T12:19:23.403Z]     at /Users/francesco.grauso/Desktop/Firma con IO/io-functions-sign/node_modules/fp-ts/lib/function.js:184:30
[2022-08-01T12:19:23.403Z]     at /Users/francesco.grauso/Desktop/Firma con IO/io-functions-sign/node_modules/fp-ts/lib/ReaderT.js:18:87
[2022-08-01T12:19:23.403Z]     at /Users/francesco.grauso/Desktop/Firma con IO/io-functions-sign/node_modules/fp-ts/lib/ReaderT.js:18:87.
```

**Pull request merged**